File: interface/data/roles/CreateRoleOptions.ts
// Copyright © Spatial Corporation. All rights reserved.

import { CreateResourceOptions } from "..";

/**
 * Configurable options for a new role.
 */
export type CreateRoleOptions = CreateResourceOptions & {
  /**
   * The name of the role.
   */
  name: string;

  /**
   * A color associated with the role.
   */
  color: string;

  /**
   * A message describing what the role does.
   */
  description?: string;
};

File: cloud/Server/Data/Roles/CreateRoleOptions.cs
// Copyright © Spatial Corporation. All rights reserved.

namespace Spatial.Cloud.Data.Roles;

/// <summary>
/// Configurable options for a new role.
/// </summary>
public class CreateRoleOptions : CreateResourceOptions
{
    /// <summary>
    /// The name of the role.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// The color associated with the role.
    /// </summary>
    public string Color { get; set; }

    /// <summary>
    /// A brief description of the role.
    /// </summary>
    public string? Description { get; set; }
}